### PR TITLE
Dependencies update

### PR DESCRIPTION
**Pull request by JulesGuesnon**
Hello !

First of all thanks for your work on this template ! 

A lot of libraries evolved, so I thought this would be cool to update them.

I also removed libraries and code because I thought this would be interesting to have the bare minimum for a React app and let people freely add their libraries.

If you think some of my changes are bad, tell me and I'll make a new commit

### CODE REVIEW

File: template.json [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
+          "eject": "react-scripts eject",
+          "re:build": "rescript build",
+          "re:watch": "rescript build -w",
+          "re:clean": "rescript clean",
+          "postinstall": "rescript build",
+          "prebuild": "rescript clean && rescript build"
+      },
+      "devDependencies": {
+          "rescript": "^9.1.2"
+      },
```
Comment (cometkim) [inline template.json]: Not using `devDependencies` is a convention in a CRA environment. please move rescript to `dependencies`

File: template.json [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
-      "npm-run-all": "^4.1.5",
-      "react": "experimental",
-      "react-dom": "experimental",
-      "react-scripts": "^4.0.1",
-      "reason-react": "^0.9.1",
-      "web-vitals": "^1.0.1"
-    },
-    "eslintConfig": null
+      "scripts": {
+          "start": "react-scripts start",
```
Comment (cometkim) [inline template.json]: This template uses a concurrent script runner to keep the experience that `yarn start` starts dev server out-of-box. any reason to remove it?

File: template/src/App.res [hunk 3]
```diff
@@ -1,9 +1,8 @@
-%%raw(`import './App.css';`)
-
-@bs.module("./logo.svg") external logo: string = "default"
+@module external _css: unit = "./App.css"
```
Comment (cometkim) [inline template/src/App.res:1]: :+1:

File: template/bsconfig.json [hunk 4]
```diff
@@ -1,30 +1,21 @@
 {
   "name": "my-rescript-app",
   "reason": {
-    "react-jsx": 3
+      "react-jsx": 3
```
Comment (cometkim) [inline template/bsconfig.json:4]: tab sizes in json files have been changed unnecessarily. (`package.json` and `bsconfig.json`)

File: template/bsconfig.json [hunk 5]
```diff
@@ -1,30 +1,21 @@
...
-    "-bs-no-version-header"
-  ],
+  "bsc-flags": ["-bs-super-errors", "-bs-no-version-header"],
   "package-specs": [
-    {
-      "module": "es6",
-      "in-source": true
-    }
+      {
+          "module": "commonjs",
```
Comment (cometkim) [inline template/bsconfig.json:13]: Why do we need to prefer commonjs output over esm?

**Review by cometkim — CHANGES_REQUESTED** [review]

File: template.json (hunk 2 above)
Comment (JulesGuesnon) [reply template.json]: I made a bunch of copy/paste and forgot to add it again, I'll add it

File: template/bsconfig.json (hunk 5 above)
Comment (JulesGuesnon) [reply template/bsconfig.json:13]: In my case, while I was trying to setup the template and verify if it was working, I couldn't get it to compile unless module was set to `commonjs`. To be honest I looked for the reason, but I have no idea why it wasn't working
But maybe it was specific to my setup and I can put back `es6`

**Comment by JulesGuesnon** [comment]
Hello ! I'm sorry, I've been very busy the last few days, I finally corrected what you asked for, let me know if any other changes are needed

**Review by tb01923 — COMMENTED** [review]
I tried running this and the ".gitignore" needed to be renamed to "gitignore" in order for the template to work.  I don't know if this is my ignorance - but I wanted to dorp a note. (installing off a local copy of the repo)

**Pull request merged**